content/tutorial-plotting-fractals.md

```diff
@@ -0,0 +1,553 @@
+
+Today we will learn how to plot these beautiful visualizations and will start to do a bit of exploring for ourselves as we gain familiarity of the mathematics behind fractals and will use the ever powerful NumPy universal functions to perform the necessary calculations efficiently.
+
++++
+
+## What you'll do
+
+- Write a function for computing Mandelbrot fractals
+- Write a function that computes Newton fractals 
+- Experiment with variations of general fractal types
+
++++
+
+## What you'll learn
+
+- A better intuition for how fractals work mathematically 
+- A basic understanding about NumPy Universal Functions and Boolean Indexing
+- The basics of working with complex numbers in NumPy 
+- How to create your own unique fractal visualizations
+
++++
+
+## What you'll need
+
+- [Matplotlib](https://matplotlib.org/)
+- make_axis_locatable function from mpl_toolkits API
+ 
+which can be imported as follows:
+
+```{code-cell} ipython3
+import numpy as np
+import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+```
+
+- Some familiarity with Python, NumPy and matplotlib
+- An idea of elementary mathematical functions, such as [exponents](https://en.wikipedia.org/wiki/Exponential_function), [sin](https://en.wikipedia.org/wiki/Sine), [polynomials](https://en.wikipedia.org/wiki/Polynomial) etc
+- A very basic understanding of [complex numbers](https://en.wikipedia.org/wiki/Complex_number) would be useful
+- Knowledge of [derivatives](https://en.wikipedia.org/wiki/Derivative) may be helpful
+
++++
+
+## Warmup
+
+To gain some intuition for what fractals are, we will begin with an example.
+
+Consider the following equation: 
+
+$f(z) = z^2 -1 $ 
+
+where z is a complex number (i.e of the form $a + bi$ )
+
+For our convenience, we will write a Python function for it
+
+```{code-cell} ipython3
+def f(z):
+    return np.square(z) - 1  
+```
+
+Note that the square function we used is an example of a **[NumPy Universal Function](https://numpy.org/doc/stable/reference/ufuncs.html)**; we will come back to the significance of this decision shortly. 
+
+To gain some intuition for the behaviour of the function, we can try plugging in some different values.
+
+For z = 0, we would expect to get -1
+
+```{code-cell} ipython3
+f(0)
+```
+
+Some values grow, some values shrink, some don't experience much change.
+
+```{code-cell} ipython3
+z = [4, 1-0.2j, 1.6]
+f(z)
+```
```

Comment:
I think it would be useful to note quickly that this is the same as applying the function to the three values separately; otherwise it might be hard to understand what we should be noticing in the values here.

Reply:
Good insight, I will incorporate this into my next update.